crates/storage/provider/src/test_utils/blocks.rs

```diff
@@ -222,3 +237,237 @@ fn block2(
 
     (SealedBlockWithSenders { block, senders: vec![Address::new([0x31; 20])] }, bundle)
 }
+
+/// Block three that points to block 2
+fn block3(
+    number: BlockNumber,
+    parent_hash: B256,
+    prev_state: &BundleStateWithReceipts,
+) -> (SealedBlockWithSenders, BundleStateWithReceipts) {
+    let address_range = 1..=20;
+    let slot_range = 1..=100;
+
```

Comment:
One note is that it should probably be much easier to create tests like this. Better utilities for creating chains of blocks with user-provided bundlestate, root, receipts, all lined up with proper hashes and parent hashes would be useful. Will create an issue on this